src/dlgPackageExporter.cpp

```diff
@@ -262,54 +461,193 @@ void dlgPackageExporter::slot_export_package()
+                }
             }
-            QString fullName = tempDir + "/" + contents[i];
-            struct zip_source* s = zip_source_file(archive, fullName.toStdString().c_str(), 0, 0);
-            if (s == nullptr) {
-                int sep = 0;
-                zip_error_get(archive, &err, &sep);
-                zip_error_to_str(buf, sizeof(buf), err, errno);
-                //FIXME: report error to userqDebug()<<"zip source error"<<fullName<<fname<<buf;
+
+#if Q_OS_WIN32
+            qt_ntfs_permission_lookup--;
+#endif
+
+            if (directoryEntries.count() > 1) {
+                std::sort(directoryEntries.begin(), directoryEntries.end());
             }
-            err = zip_add(archive, fname.toStdString().c_str(), s);
-            if (err == -1) {
-                int sep = 0;
-                zip_error_get(archive, &err, &sep);
-                zip_error_to_str(buf, sizeof(buf), err, errno);
-                //FIXME: report error to userqDebug()<<"added file error"<<fullName<<fname<<buf;
+
+            QStringListIterator itDirectoryName(directoryEntries);
+            while (itDirectoryName.hasNext() && isOk) {
+                QString directoryName = itDirectoryName.next();
+                if (zip_dir_add(archive, directoryName.toStdString().c_str(), ZIP_FL_ENC_UTF_8) == -1) {
+                    // zip_dir_add(...) returns the index of this item in the
+                    // archive or -1 on error
+
+                    infoLabel->setText(QStringLiteral("<font color='red'><big><b>%1</b></big></font>")
+                                       .arg(tr("Failed to add directory \"%1\" to module (archive) file \"%2\", error message was:\n"
+                                               "%3").arg(directoryName, mZipFile, QString::fromUtf8(zip_strerror(archive)))));
+                    zip_close(archive);
+                    isOk = false;
+                }
             }
-        }
-        err = zip_close(archive);
-        if (err != 0) {
-            zip_error_to_str(buf, sizeof(buf), err, errno);
-            //FIXME: report error to userqDebug()<<"dp close file error"<<buf;
-            close();
-            return;
-        }
+
+            // Process the config and the file containing troleshe Mudlet triggers,
+            // etc specially so they are inserted first and last respectively:
+            if (isOk) {
+                if (fileEntries.contains(QStringLiteral("config.lua"))) {
+                    if (!writeFileToZip(QStringLiteral("config.lua"), fileEntries.value(QStringLiteral("config.lua")), archive)) {
+                        isOk = false;
+                    }
+                    fileEntries.remove(QStringLiteral("config.lua"));
+                } else {
+                    qWarning() << "dlgPackageExporter::slot_exportPackage() - ERROR: failed to add package detail file config.lua to archive, the package/module may not work!";
```

Comment:
It'll work just fine without it, the name will just be assumed from the zip file instead.

Reply:
... but it is a warning that the `config.lua` file has been clobbered...

Reply:
Yeah, but the text is incorrect.

Reply:
Would:

    qWarning() << "dlgPackageExporter::slot_exportPackage() - WARNING: failed to add package detail file config.lua to archive!"

be acceptable?

Reply:
:+1:

Reply:
☑️ Done